core/src/Kokkos_View.hpp

```diff
@@ -557,13 +567,15 @@ class View : public Impl::BasicViewFromTraits<DataType, Properties...>::type {
           std::is_constructible_v<
               mdspan_type, typename View<OtherT, OtherArgs...>::mdspan_type>,
           void*> = nullptr)
-      : base_t(static_cast<typename mdspan_type::data_handle_type>(
-                   other.data_handle()),
-               static_cast<typename mdspan_type::mapping_type>(other.mapping()),
-               static_cast<typename mdspan_type::accessor_type>(
-                   other.accessor())) {
-    base_t::check_basic_view_constructibility(other.mapping());
-  }
+      : base_t([&]() {
+          base_t::check_basic_view_constructibility(other.mapping());
+          return base_t(
+              static_cast<typename mdspan_type::data_handle_type>(
+                  other.data_handle()),
+              static_cast<typename mdspan_type::mapping_type>(other.mapping()),
+              static_cast<typename mdspan_type::accessor_type>(
+                  other.accessor()));
+        }()) {}
```

Comment:
I would have gone for a comma fold, but I am ok with an immediately invoked lambda.
Please add a comment that we check the precondition before invoking the mdspan mapping constructor so we can diagnose the precondition violation and produce a good quality error message

Reply:
... which we would not to be able to do if the mdspan implementation enforce the contract

Reply:
Looks like I had pushed to the wrong branch.